source/code/plugins/oms_common.rb

```diff
@@ -478,6 +480,141 @@ class Common
+        return false
+      end
+
+      def is_recognized_prefix_label?(fqdnLabel)
+        s = fqdnLabel.strip.downcase
+        return true if s == 'www'
+        return false
+      end
+
+      def look_for_sockety_host_address
+        hnb = nil
+
+        begin
+          socket_hostname = Socket.gethostname
+          hnb = socket_hostname.strip
+        rescue => error
+          OMS::Log.error_once("Unable to get the Host Name using socket facility: #{error}")
+          return
+        end
+
+        if is_dot_separated_string?(hnb) then
+          if is_like_ipv4_string?(hnb) then
+            # If Socket.gethostname returns an IPv4 address,
+            # use the entire string.
+            @@Hostname = hnb
+          else
+            # Here we presume we got a fully qualified domain name.
+            labelarray = hnb.split(".")
+            if is_recognized_prefix_label?(labelarray[0]) then
+              # Presume first label is a catagory aspect like www, and use the second.
+              @@Hostname = clean_hostname_string(labelarray[1])
+            else
+              # Presume first is the one.
+              @@Hostname = clean_hostname_string(labelarray.first)
+            end
+          end
+        else
+          # Here we presume we got a usable hostname,
+          # though it could also be an IPV6, or other things,
+          # and more attention may be needed here later.
+          @@Hostname = clean_hostname_string(hnb)
+        end
+      end
+
+      def look_in_Designated_HostnameFile
+        # Issue:
+        #   When omsagent runs inside a container, gethostname returns the hostname of the container (random name)
+        #   not the actual machine hostname.
+        #   One way to solve this problem is to set the container hostname same as machine name, but this is not
+        #   possible when host-machine is a private VM inside a cluster.
+        # Solution:
+        #   Share/mount ‘/etc/hostname’ as '/var/opt/microsoft/omsagent/state/containername' with container and
+        #   omsagent will read hostname from shared file.
+
+        unless File.readable?(@@HostnameFilePath)
+            OMS::Log.warn_once("File '#{@@HostnameFilePath}' exists but is not readable.")
+            return
+        end
+
+        begin
+          hnb = File.read(@@HostnameFilePath)
+          @@Hostname = clean_hostname_string(hnb)
+        rescue => error
+          OMS::Log.warn_once("Unable to read the hostname from #{@@HostnameFilePath}: #{error}")
+        end
+
+      end
+
+      def validate_hostname_equivalent(hnBuffer)
+        # RFC 2181
+        rfcl = "RFC 2181 specified hostname range of {1,63} octets for non-root item."
+        return if is_hostname_compliant?(hnBuffer)
+        return if is_like_ipv4_string?(hnBuffer) 
+        return if is_like_ipv6_string?(hnBuffer)
+        msg = "Hostname '#{hnBuffer}' length of #{hnBuffer} out of range (#{rfcl})."
```

Comment:
You print hnBuffer twice here, and not the length of hnBuffer.

Reply:
Got it.